### ISSUE TITLE
Help on implementation of LTMU trackers on your framework

### ISSUE THREAD

**Issue by johnSmith1990** [visionml/pytracking#245]
Hi all, Is there any solution for LTMU [()trackers](https://github.com/Daikenan/LTMU/tree/master/PrDiMP_MU) in which we can test on your framework?
As i tried, they only works on sequences with gt files, and don't test on video or camera.

regards

**Comment by martin-danelljan** [comment]
Hi. It should be perfectly possible. But I do not know what changes the LTMU authors made to pytracking when building their trackers. Since they use a deep copy of the full repository for each new tracker, I expect that there are some larger changes to the machinery.

Our framework is very general and it should be not difficult to integrate LTMU in a clean way. But you need to ask the authors or figure out how to do that. I can only help you with things in the pytracking framework.

/Martin

**Comment by johnSmith1990** [comment]
Thanks for your supports.
LTMU authors don't answer. But as I see the codes, the tracker classes differ from your implementation/ for example you can see dimp class. Anyway, The LTMU is an extension for Tracker and It is not a Tracke, so can add to any tracker. I have to implement that tracker for the webcam. I am reading your and LTMU codes. Any help would be much appreciated.

**Comment by martin-danelljan** [comment]
Yeah. So, we already have a few  different trackers and segmentation methods implemented (ECO, ATOM, DiMP, LWL, etc). So there are plenty of examples how to integrate a new tracker in our framework. Unfortunately the LTMU authors did not follow this convention of how it was "intended" to integrate a new tracker. So I can only help with questions regarding our code and not about what LTMU authors did.

I can also mention that we will in near future release code for a new tracker https://arxiv.org/abs/2103.16556 that outperforms LTMU. So you also have the option to wait for that :)

**Comment by johnSmith1990** [comment]
Thanks for your response dear Martin. I am a developer and I prefer to improve my implementation skill as well. If you help me with that, I can start implementing Any Tracker on your framework and share it with you. Another wodrs, I need some implementation project in this field(Tracking) rather than source code. But I need some hints to start and go step by step( you have great experience and skills). Thank you so much.

**Comment by johnSmith1990** [comment]
It is a great work((https://arxiv.org/abs/2103.16556) 
I want to implement it as a project :cry:

**Comment by lfxx** [comment]
> Yeah. So, we already have a few different trackers and segmentation methods implemented (ECO, ATOM, DiMP, LWL, etc). So there are plenty of examples how to integrate a new tracker in our framework. Unfortunately the LTMU authors did not follow this convention of how it was "intended" to integrate a new tracker. So I can only help with questions regarding our code and not about what LTMU authors did.
> 
> I can also mention that we will in near future release code for a new tracker [arxiv.org/abs/2103.16556](https://arxiv.org/abs/2103.16556) that outperforms LTMU. So you also have the option to wait for that :)

Any update plan for your new model?can't wait to see it.

**Comment by 2006pmach** [comment]
> > Yeah. So, we already have a few different trackers and segmentation methods implemented (ECO, ATOM, DiMP, LWL, etc). So there are plenty of examples how to integrate a new tracker in our framework. Unfortunately the LTMU authors did not follow this convention of how it was "intended" to integrate a new tracker. So I can only help with questions regarding our code and not about what LTMU authors did.
> > I can also mention that we will in near future release code for a new tracker [arxiv.org/abs/2103.16556](https://arxiv.org/abs/2103.16556) that outperforms LTMU. So you also have the option to wait for that :)
> 
> Any update plan for your excellent new tracker?can't wait to see it.Hope for a obvious improvement on long-term tracking.

We just released the code for KeepTrack. Have fun and thanks for your interest.
